festim/settings.py

```diff
@@ -0,0 +1,53 @@
+import festim as F
+
+
+class Settings:
+    """Settings for a festim simulation.
+
+    Args:
+        atol (float): Absolute tolerance for the solver.
+        rtol (float): Relative tolerance for the solver.
+        max_iterations (int, optional): Maximum number of iterations for the
+            solver. Defaults to 30.
+        final_time (float, optional): Final time for a transient simulation.
+            Defaults to None
+        stepsize (festim.Stepsize, optional): stepsize for a transient
+            simulation. Defaults to None
+
+    Attributes:
+        atol (float): Absolute tolerance for the solver.
+        rtol (float): Relative tolerance for the solver.
+        max_iterations (int, optional): Maximum number of iterations for the solver.
+        final_time (float, optional): Final time for a transient simulation.
+        stepsize (festim.Stepsize, optional): stepsize for a transient
```

Comment:
Maybe just a copy-paste issue but:

Attributes cannot be optional. They may have a default value, but that doesn't make them optional.


```suggestion
        max_iterations (int): Maximum number of iterations for the solver.
        final_time (float): Final time for a transient simulation.
        stepsize (festim.Stepsize): stepsize for a transient
```